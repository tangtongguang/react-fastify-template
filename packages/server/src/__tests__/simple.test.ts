import fastify, { FastifyInstance } from "fastify"
import { IncomingMessage, Server, ServerResponse } from "http"
import statusRoutes from '../modules/routes/status'
describe("/status", () => {
  let server: FastifyInstance<Server, IncomingMessage, ServerResponse>;

  beforeAll(() => { })

  beforeEach(async () => {
    server = fastify({})
    server.register(statusRoutes)
    await server.ready();
    jest.clearAllMocks()
  })


  it('GET return 200', async done => {
    const response = await server.inject({ method: 'GET', url: '/status' });
    expect(response.statusCode).toEqual(200);
    const payload: { works: boolean } = JSON.parse(response.payload)
    expect(payload).toMatchSnapshot({ works: false })
    done();
  })
})