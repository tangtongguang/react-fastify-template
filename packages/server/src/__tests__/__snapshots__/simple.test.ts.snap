// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/status GET return 200 1`] = `
Object {
  "works": false,
}
`;
